test/backend-test/test-mqtt.js

```diff
@@ -100,4 +100,46 @@ describe("MqttMonitorType", {
             new Error("Message received but value is not equal to expected value, value was: [present]")
         );
     });
+
+    test('should match exact topic without wildcards', () => {
+        const regex = MqttMonitorType.mqttTopicRegex('sensor/temperature');
+        assert.ok(regex.test('sensor/temperature') === true);
+        assert.ok(regex.test('sensor/humidity') === false);
+    });
+
+    test('should match exact topic without wildcards but with special characters', () => {
+        const regex = MqttMonitorType.mqttTopicRegex('sensor.pomme/temperature');
+        assert.ok(regex.test('sensor.pomme/temperature') === true);
+        assert.ok(regex.test('sensor.pomme/humidity') === false);
+    });
+
```

Comment:
please add a testcase specifically for the escaping of `#`, `+` and the other regex characters.